src/sentry/integrations/opsgenie/actions/notification.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+import logging
+
+from sentry.integrations.opsgenie.actions import OpsgenieNotifyTeamForm
+from sentry.integrations.opsgenie.client import OpsgenieClient
+from sentry.rules.actions import IntegrationEventAction
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.services.hybrid_cloud.integration.model import RpcOrganizationIntegration
+from sentry.shared_integrations.client.proxy import infer_org_integration
+from sentry.shared_integrations.exceptions import ApiError
+
+logger = logging.getLogger("sentry.integrations.opsgenie")
+
+
+class OpsgenieNotifyTeamAction(IntegrationEventAction):
+    id = "sentry.integrations.opsgenie.notify_action.OpsgenieNotifyTeamAction"
+    form_cls = OpsgenieNotifyTeamForm
+    label = "Send a notification to Opsgenie account {account} and team {team}"
+    prompt = "Send an Opsgenie notification"
+    provider = "opsgenie"
+    integration_key = "account"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.form_fields = {
+            "account": {
+                "type": "choice",
+                "choices": [(i.id, i.name) for i in self.get_integrations()],
+            },
+            "team": {"type": "choice", "choices": self.get_teams()},
+        }
+
+    def _get_team(self, org_integration: RpcOrganizationIntegration | None):
+        if not org_integration:
+            return None
+        teams = org_integration.config.get("team_table")
+        if not teams:
+            return None
+        team_id = self.get_option("team")
+        for team in teams:
+            if team["id"] == team_id:
+                return team
+        return None
```

Comment:
Maybe we can look at reusing this method for the form field validation actually? I know the validation accomplishes essentially the same thing, but if they were to drift, what we actually care about is that the alert can fetch a team from the org_integration, not that the team exists on some org_integration